lib/spree_multi_domain/multi_domain_helpers.rb

```diff
@@ -25,7 +25,7 @@ def add_current_store_id_to_params
 
     private
     def store_key
-      request.headers['HTTP_SPREE_STORE'] || request.env['SERVER_NAME']
+      current_store_class.new(request).send(:store_key)
```

Comment:
good point, I hadn't looked too closely at the `Spree::Tracker.current`. it looks like it duplicates some of `Spree::Store.current(store_key)`.  I'll fix